pkg/scheduler/framework/framework.go

```diff
@@ -662,26 +663,53 @@ func (f *framework) manipulateBindings(
 
 // createBindings creates a list of new bindings.
 func (f *framework) createBindings(ctx context.Context, toCreate []*placementv1beta1.ClusterResourceBinding) error {
+	// issue all the create requests in parallel
+	errs, cctx := errgroup.WithContext(ctx)
 	for _, binding := range toCreate {
-		// TO-DO (chenyu1): Add some jitters here to avoid swarming the API when there is a large number of
-		// bindings to create.
-		if err := f.client.Create(ctx, binding); err != nil {
-			return controller.NewCreateIgnoreAlreadyExistError(fmt.Errorf("failed to create binding %s: %w", binding.Name, err))
-		}
+		newBinding := binding
+		errs.Go(func() error {
+			return retry.OnError(retry.DefaultBackoff,
+				func(err error) bool {
+					return apierrors.IsServiceUnavailable(err) || apierrors.IsServerTimeout(err)
+				},
+				func() error {
+					err := f.client.Create(cctx, newBinding)
+					if err != nil {
+						if apierrors.IsAlreadyExists(err) {
+							// The binding already exists, which is fine.
+							return nil
+						}
+						klog.ErrorS(err, "failed to create a new binding", "clusterResourceBinding", klog.KObj(newBinding))
+					}
+					return err
+				})
+		})
 	}
-	return nil
+	return errs.Wait()
 }
 
 // patchBindings patches a list of existing bindings using JSON patch.
 func (f *framework) patchBindings(ctx context.Context, toPatch []*bindingWithPatch) error {
-	// TODO (rzhang): issue those patches in parallel, retry if there is conflict
+	// issue all the patch requests in parallel
+	errs, cctx := errgroup.WithContext(ctx)
 	for _, bp := range toPatch {
-		// Use JSON patch to avoid races.
-		if err := f.client.Patch(ctx, bp.updated, bp.patch); err != nil {
-			return controller.NewUpdateIgnoreConflictError(fmt.Errorf("failed to patch binding %s: %w", bp.updated.Name, err))
-		}
+		patchBinding := bp
+		errs.Go(func() error {
+			return retry.OnError(retry.DefaultBackoff,
+				func(err error) bool {
+					return apierrors.IsServiceUnavailable(err) || apierrors.IsServerTimeout(err)
+				},
+				func() error {
+					// Use JSON patch to avoid races so we shouldn't get a conflict error.
+					err := f.client.Patch(cctx, patchBinding.updated, patchBinding.patch)
+					if err != nil {
+						klog.ErrorS(err, "failed to patch a binding", "clusterResourceBinding", klog.KObj(patchBinding.updated))
```

Comment:
klog.ErrorS(err, "Failed to patch a binding", "clusterResourceBinding", klog.KObj(patchBinding.updated))